frontend/catalyst/from_plxpr.py

```diff
@@ -351,7 +356,27 @@ def interpret_measurement_eqn(self, eqn: jax.core.JaxprEqn):
         )[0]
 
         primitive = measurement_map[eqn.primitive.name]
-        mval = primitive.bind(obs, shape=shaped_array.shape, shots=self._device.shots.total_shots)
+        device_shots = get_device_shots(self._device) or 0
+
+        # TODO: as we are in the process of migrating to dynamic measurement primitive shapes,
+        # we will gradually get rid of the shape argument for these primitives
+        # While we are in the migrating process, we need to handle them explicitly one by one
+        if primitive is sample_p:
+            mval = (
+                primitive.bind(obs, shots=device_shots, num_qubits=shaped_array.shape[1])
+                if isinstance(device_shots, int)
+                else primitive.bind(obs, device_shots, num_qubits=shaped_array.shape[1])
+            )
```

Comment:
Good idea!

Maybe we can even have some sort of utility interface like 
```
def bind_primitive(p, static_args, dyn_args):
   p.bind(dyn_args, static_agrs=static_agrs)
```
since we are doing other things with static/dynamic too, but probably that's for a another PR. Here I'll just outline this shots binding.

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1310/commits/2b1aa2222b3f421e0f80e33b2a7aa89dbb6961b2

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1310/commits/471c99d183474caedb311e403274cc36b41d6919